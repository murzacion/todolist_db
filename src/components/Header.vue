<template>
  <v-app-bar app>
    <v-toolbar-title>ToDoList</v-toolbar-title>
    <v-spacer></v-spacer>
    <v-toolbar-title class="mr-2 NumberTasks"
      >{{ NumberTasks > 1 ? msg2 : msg1 }}:</v-toolbar-title
    >
    <v-progress-circular
      :value="progress"
      :size="50"
      :width="5"
      style="font-size: 20px"
      class="mr-2"
    >
      {{ NumberTasks }}
    </v-progress-circular>
  </v-app-bar>
</template>

<script>
export default {
  data() {
    return {
      msg1: "Task",
      msg2: "Tasks"
    };
  },
  computed: {
    completedTasks() {
      return this.$store.getters.selectedTask.length;
    },
    progress() {
      return (this.completedTasks / this.$store.getters.Tasks.length) * 100;
    },
    remainingTasks() {
      return this.$store.getters.Tasks.length - this.completedTasks;
    },
    NumberTasks() {
      return this.$store.getters.numberOfTasks;
    }
  }
};
</script>
<style lang="scss">
.NumberTasks {
  font-size: 1.25rem;
}
</style>
