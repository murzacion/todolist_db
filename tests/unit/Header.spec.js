import Vue from "vue";
import Vuetify from "vuetify";
Vue.use(Vuetify);
import Header from "@/components/Header.vue";
import { createLocalVue, shallowMount } from "@vue/test-utils";

const localVue = createLocalVue();

describe("Header.vue", () => {
  let vuetify;
  beforeEach(() => {
    vuetify = new Vuetify();
  });
  test("snapshot", () => {
    const $store = {
      state: {
        tasks: [],
        NumberTasks: 0
      },
      getters: {
        numberOfTasks: 0,
        Tasks: [],
        selectedTask: []
      }
    };
    const wrapper = shallowMount(Header, {
      localVue,
      vuetify,
      mocks: {
        $store
      }
    });
    expect(wrapper).toMatchSnapshot();
  });
});
