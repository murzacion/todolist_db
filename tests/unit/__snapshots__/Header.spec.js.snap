// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.vue snapshot 1`] = `
"ToDoList  Task: 
    () => ({})"
`;
